test/palindrome.bats

```diff
@@ -0,0 +1,515 @@
+setup() {
+    load 'test_helper/bats-support/load'
+    load 'test_helper/bats-assert/load'
+    # ... the remaining setup is unchanged
+
+    # get the containing directory of this file
+    # use $BATS_TEST_FILENAME instead of ${BASH_SOURCE[0]} or $0,
+    # as those will point to the bats executable's location or the preprocessed file respectively
+    DIR="$( cd "$( dirname "$BATS_TEST_FILENAME" )" >/dev/null 2>&1 && pwd )"
+    # make executables in src/ visible to PATH
+    PATH="$DIR/../src:$PATH"
+}
+
+@test "'' is palindrome" {
+    run bash -c './_build/default/bin/turing our-machines/palindrome.json "" | cat -e'
+    assert_output 'Interpreter starting...$
+Tape: [<".">] ("pick_character", ".") -> ("write_is_palindrome", ".", right)$
+Tape: [".",<".">] ("write_is_palindrome", ".") -> ("HALT", "y", right)$
+Tape: [".","y",<".">] Final state reached !$
+Interpreter closing...$'
+}
+
+@test "'a' is palindrome" {
+    run bash -c './_build/default/bin/turing our-machines/palindrome.json "a" | cat -e'
+    assert_output 'Interpreter starting...$
+Tape: [<"a">] ("pick_character", "a") -> ("go_to_end_and_find_a", ".", right)$
+Tape: [".",<".">] ("go_to_end_and_find_a", ".") -> ("is_a", ".", left)$
+Tape: [<".">,"."] ("is_a", ".") -> ("write_is_palindrome", ".", right)$
+Tape: [".",<".">] ("write_is_palindrome", ".") -> ("HALT", "y", right)$
+Tape: [".","y",<".">] Final state reached !$
+Interpreter closing...$'
+}
+
+@test "'aa' is palindrome" {
+    run bash -c './_build/default/bin/turing our-machines/palindrome.json "aa" | cat -e'
+    assert_output 'Interpreter starting...$
+Tape: [<"a">,"a"] ("pick_character", "a") -> ("go_to_end_and_find_a", ".", right)$
+Tape: [".",<"a">] ("go_to_end_and_find_a", "a") -> ("go_to_end_and_find_a", "a", right)$
+Tape: [".","a",<".">] ("go_to_end_and_find_a", ".") -> ("is_a", ".", left)$
+Tape: [".",<"a">,"."] ("is_a", "a") -> ("go_to_beginning", ".", left)$
+Tape: [<".">,".","."] ("go_to_beginning", ".") -> ("pick_character", ".", right)$
+Tape: [".",<".">,"."] ("pick_character", ".") -> ("write_is_palindrome", ".", right)$
+Tape: [".",".",<".">] ("write_is_palindrome", ".") -> ("HALT", "y", right)$
+Tape: [".",".","y",<".">] Final state reached !$
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought:** I had the same use case, this totally avoidable but it would quite easier to read the machine if after writing the y or n going to left, it will avoid to extend the tape

Reply:
You're right, I will update the machine.
Although, I think the real issue is that the format of our tape is too verbose.

Instead of `Tape: ["x", "x"] ("state", "x") -> ("state", "x", right)`

We could write: `[xx] (state, x) -> (state, x, right)`

Reply:
Great !

Indeed this could be a nice refactor :thinking: